flyway-core/src/main/java/org/flywaydb/core/internal/dbsupport/oracle/OracleSchema.java

```diff
@@ -339,39 +353,86 @@ private boolean spatialExtensionsAvailable() throws SQLException {
 
     @Override
     protected Table[] doAllTables() throws SQLException {
-        List<String> tableNames = jdbcTemplate.queryForStringList(
-                // For every table this query will count the number of references (including the transitive ones)
-                // and order the result list using that value.
-                " SELECT r FROM" +
-                        "   (SELECT CONNECT_BY_ROOT t r FROM" +
-                        "     (SELECT DISTINCT c1.table_name f, NVL(c2.table_name, at.table_name) t" +
-                        "     FROM all_constraints c1" +
-                        "       RIGHT JOIN all_constraints c2 ON c2.constraint_name = c1.r_constraint_name" +
-                        "       RIGHT JOIN all_tables at ON at.table_name = c2.table_name" +
-                        "     WHERE at.owner = ?" +
+
+        List<String> tableNames;
+
+        if (majorVersion >= 11)
```

Comment:
I would also check if all_part_tables has any row for the specified schema with reference partitioning type. This feature is used quite rare, so why to do excessive joins and perform hierarchical query in vain?